homeassistant/helpers/selector.py

```diff
@@ -31,16 +55,16 @@ def _get_selector_class(config: Any) -> type[Selector]:
     return selector_class
 
 
-def selector(config: Any) -> Selector:
+def selector(selector_type: SelectorType, config: Any) -> Selector:
```

Comment:
We could use overloading here to make it more type aware:

e.g.

```suggestion
@overload
def selector(
    selector_type: Literal[SelectorType.ACTION], config: ActionSelectorDict
) -> Selector:
    ...


@overload
def selector(
    selector_type: Literal[SelectorType.ENTITY], config: EntitySelectorDict
) -> Selector:
    ...

def selector(selector_type: SelectorType, config: Any) -> Selector:
```

Reply:
great suggestion thanks!